rlclientlib/logger/async_batcher.h

```diff
@@ -151,11 +158,43 @@ namespace reinforcement_learning { namespace logger {
       }
     }
 
-    RETURN_IF_FAIL(collection_serializer.finalize(status));
-
+    if (_events_counter_status == events_counter_status::ENABLE) {
+      unsigned int buffer_start_event_number;
+      unsigned int original_count;
+      buffer_start_event_number = _buffer_end_event_number;
+      _buffer_end_event_number = evt.get_number_of_events();
+      original_count = (_buffer_end_event_number - buffer_start_event_number);
+      if (_queue.size() == 0) {
```

Comment:
not sure about this part:
1) for high traffic loop (which we are targeting with this "if") _queue.size() will never be 0 so the trick probably will never work
2) reset_counter_get_count() is obtaining lock
Maybe we can just make this counter uint64_t?

Reply:
I'll agree here, with the counter as uint64 we will most likely restart any rlclientlib instance before overflowing the counter

Reply:
limit is ~ 10^19. so for 1 million RPS setting we have 10^19 / (365 * 24 * 3600) > 10^11 years until restart